lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -234,6 +230,16 @@ public function hasMetadataFor($className)
     public function setMetadataFor($className, $class)
     {
         $this->loadedMetadata[$className] = $class;
+
+        if (! $this->cacheDriver) {
```

Comment:
```suggestion
        if ($this->cacheDriver === null) {
```
Handling this in a rebase too.